localstack/services/apigateway/provider.py

```diff
@@ -110,28 +114,48 @@ def on_after_init(self):
     def test_invoke_method(
         self, context: RequestContext, request: TestInvokeMethodRequest
     ) -> TestInvokeMethodResponse:
-
         invocation_context = to_invocation_context(context.request)
         invocation_context.method = request["httpMethod"]
+        invocation_context.api_id = request["restApiId"]
+        invocation_context.path_with_query_string = request["pathWithQueryString"]
+
+        response = connect_to().apigateway.get_resource(
+            restApiId=invocation_context.api_id, resourceId=request["resourceId"]
+        )
```

Comment:
It should return a 404 in case the resource doesn't exist, and given the api id is a required parameter it should fail on the client side (using aws clients). However for the rest api case we can make sure it's present and it's in our store.

Good point accessing moto or our store for the resource directly would be a better way to do it.